### PR TITLE
Allow adding comma-separated monitor list

### PR DESCRIPTION
**Pull request by tgulacsi**
This helps adding the same herd of monitors to similar bugs: just copy
the list, comma-separated into the "username" field.

### CODE REVIEW

File: bug_monitor_add.php [hunk 1]
```diff
@@ -55,26 +55,32 @@
...
-	$t_user_id = $t_logged_in_user_id;
+if( is_blank( $f_usernames ) ) {
+	$t_user_ids = array( $t_logged_in_user_id );
 } else {
-	$t_user_id = user_get_id_by_name( $f_username );
-	if( $t_user_id === false ) {
-		$t_user_id = user_get_id_by_realname( $f_username );
-
+	$t_user_ids = array();
+	foreach( explode( ',', $f_usernames ) as $f_username ) {
```
Comment (vboctor) [inline bug_monitor_add.php]: We should support splitting based on ',' and '|'.

Comment (vboctor) [inline bug_monitor_add.php]: Don't use $f_username, use $t_username instead.  $f_ are used for form fields.

File: bug_monitor_add.php [hunk 2]
```diff
@@ -55,26 +55,32 @@
...
+		$t_user_id = user_get_id_by_name( $f_username );
```
Comment (vboctor) [inline bug_monitor_add.php]: Trip username before passing in to this method or trim() within the method.

File: bug_monitor_add.php [hunk 3]
```diff
@@ -55,26 +55,32 @@
...
+			}
 		}
+		$t_user_ids[] = $t_user_id;
 	}
 }
 
-if( user_is_anonymous( $t_user_id ) ) {
-	trigger_error( ERROR_PROTECTED_ACCOUNT, E_USER_ERROR );
+foreach( $t_user_ids as $t_user_id ) {
+	if( user_is_anonymous( $t_user_id ) ) {
```
Comment (vboctor) [inline bug_monitor_add.php]: Should this check be moved in the same loop as the access check below?

File: bug_monitor_add.php [hunk 4]
```diff
@@ -85,13 +91,15 @@
...
-}
+foreach( $t_user_ids as $t_user_id ) {
+	if( $t_logged_in_user_id == $t_user_id ) {
+		access_ensure_bug_level( config_get( 'monitor_bug_threshold' ), $f_bug_id );
+	} else {
+		access_ensure_bug_level( config_get( 'monitor_add_others_bug_threshold' ), $f_bug_id );
+	}
 
-bug_monitor( $f_bug_id, $t_user_id );
+	bug_monitor( $f_bug_id, $t_user_id );
```
Comment (vboctor) [inline bug_monitor_add.php]: Maybe have a separate loop to add all monitors.  This way if this loop fails, we end up with no changes, rather than half the changes.

File: bug_monitor_add.php [hunk 5]
```diff
@@ -85,13 +91,15 @@
...
+}
 
 form_security_purge( 'bug_monitor_add' );
```
Comment (vboctor) [inline bug_monitor_add.php]: Give the amount of possible errors above, I wonder if this call should be earlier on this page.  Not part of this change, but should be easy to change.

File: bug_monitor_add.php [hunk 6]
```diff
@@ -55,26 +55,34 @@
...
-		$t_user_id = user_get_id_by_realname( $f_username );
-
-		if( $t_user_id === false ) {
-			error_parameters( $f_username );
-			trigger_error( ERROR_USER_BY_NAME_NOT_FOUND, E_USER_ERROR );
+	$t_usernames = array( $f_usernames );
+	foreach( array( "|", "," ) as $t_sep ) {
+		if( strpos( $f_usernames, $t_sep ) !== false ) {
+			$t_usernames = explode( $t_sep, $f_usernames );
+			break;
```
Comment (dregad) [inline bug_monitor_add.php]: Wouldn't it be simpler to do 
``` php
$t_usernames = preg_split( '/[,|]/', $f_usernames, -1, PREG_SPLIT_NO_EMPTY )
```
And maybe add an `array_unique()` call too, just in case.

File: bug_monitor_add.php [hunk 7]
```diff
@@ -85,13 +93,23 @@
...
+
+	if( $t_logged_in_user_id == $t_user_id ) {
+		access_ensure_bug_level( config_get( 'monitor_bug_threshold' ), $f_bug_id );
+	} else {
+		access_ensure_bug_level( config_get( 'monitor_add_others_bug_threshold' ), $f_bug_id );
+	}
 }
 
-bug_monitor( $f_bug_id, $t_user_id );
+// then add only if all can be added.
```
Comment (dregad) [inline bug_monitor_add.php]: We use `#` for inline comments

File: bug_monitor_add.php [hunk 8]
```diff
@@ -85,13 +93,23 @@
 	$g_project_override = $t_bug->project_id;
 }
 
-if( $t_logged_in_user_id == $t_user_id ) {
-	access_ensure_bug_level( config_get( 'monitor_bug_threshold' ), $f_bug_id );
-} else {
-	access_ensure_bug_level( config_get( 'monitor_add_others_bug_threshold' ), $f_bug_id );
+// Check all monitors first,
```
Comment (dregad) [inline bug_monitor_add.php]: We use `#` for inline comments

File: bug_monitor_add.php [hunk 9]
```diff
@@ -55,26 +55,34 @@
...
+			$t_user_id = user_get_id_by_realname( $t_username );
 
-if( user_is_anonymous( $t_user_id ) ) {
-	trigger_error( ERROR_PROTECTED_ACCOUNT, E_USER_ERROR );
+			if( $t_user_id === false ) {
+				error_parameters( $t_username );
+				trigger_error( ERROR_USER_BY_NAME_NOT_FOUND, E_USER_ERROR );
+			}
+		}
+		$t_user_ids[] = $t_user_id;
```
Comment (dregad) [inline bug_monitor_add.php]: `$t_user_ids[$t_user_id] = $t_user_id;` prevents duplicates

File: bug_monitor_add.php [hunk 10]
```diff
@@ -85,13 +93,23 @@
 	$g_project_override = $t_bug->project_id;
 }
 
-if( $t_logged_in_user_id == $t_user_id ) {
-	access_ensure_bug_level( config_get( 'monitor_bug_threshold' ), $f_bug_id );
-} else {
-	access_ensure_bug_level( config_get( 'monitor_add_others_bug_threshold' ), $f_bug_id );
+// Check all monitors first,
+foreach( $t_user_ids as $t_user_id ) {
```
Comment (dregad) [inline bug_monitor_add.php]: Is there any benefit of doing these 2 steps in an extra loop ?

**Comment by dregad** [comment]
Thanks for your contribution.
Can you please reference an issue in our tracker (create one if needed.

File: bug_monitor_add.php (hunk 6 above)
Comment (tgulacsi) [inline bug_monitor_add.php]: If I'd know PHP, then yes :) So thanks for the simpler solution!

File: bug_monitor_add.php (hunk 10 above)
Comment (tgulacsi) [inline bug_monitor_add.php]: Nope, just conserved the logic of the original.

**Comment by dregad** [comment]
Merged in d7953e10ff9c0b48320448a0309e15071786003c. Thank you !

**Comment by dregad** [comment]
@tgulacsi as a side note, it would appear the e-mail address you used to commit https://github.com/mantisbt/mantisbt/commit/b694c1434ac366ed67744ca47c837518065f6005 is not registered with your GitHub account, you may want to fix that. 

2016-03-07 10_07_07-allow adding or _ separated monitor list mantisbt_mantisbt b694c14
